app/models/item.rb

```diff
@@ -62,4 +62,24 @@ def return_stock(amount)
   def sell_stock(amount)
     self.inventory -= amount
   end
+
+  def find_discounts
+   merchant.discounts
+  end
+
+  def no_discounts?(quantity)
+    merchant.discounts.where("#{quantity} >= discounts.item_amount") == []
+  end
+
+  def select_highest_discount(quantity)
+    merchant.discounts
+            .where("#{quantity} >= discounts.item_amount")
+            .order('discounts.percentage DESC')
+            .first
+  end
+
+  def apply_discount(percentage)
```

Comment:
this method removes some of the logic out of the cart controller, might make more sense to be in the cart model